ktor-server/ktor-server-core/jsAndWasmShared/src/io/ktor/server/engine/ShutdownHook.jsAndWasmShared.kt

```diff
@@ -7,16 +7,7 @@ package io.ktor.server.engine
 import io.ktor.events.*
 import kotlin.concurrent.*
 
-/**
- * Adds automatic application shutdown hooks management. Should be used **before** starting the engine.
- * Once application termination noticed, [stop] block will be executed.
- * Please note that a shutdown hook only registered when the application is running. If the application
- * is already stopped then there will be no hook and no [stop] function invocation possible.
- * So [stop] block will be called once or never.
- */
-public actual fun ApplicationEngine.addShutdownHook(monitor: Events, stop: () -> Unit) {
-    addProcessShutdownHook(stop)
-}
+internal actual val SHUTDOWN_HOOK_ENABLED = true
```

Comment:
Are there alternatives to `System.getProperty` for native and JS platforms?

Reply:
Not that I know of...